mypy/nodes.py

```diff
@@ -689,12 +712,14 @@ def __init__(
         arguments: list[Argument] | None = None,
         body: Block | None = None,
         typ: mypy.types.FunctionLike | None = None,
+        type_args: list[TypeParam] | None = None,
     ) -> None:
         super().__init__()
         self.arguments = arguments or []
         self.arg_names = [None if arg.pos_only else arg.variable.name for arg in self.arguments]
         self.arg_kinds: list[ArgKind] = [arg.kind for arg in self.arguments]
         self.max_pos: int = self.arg_kinds.count(ARG_POS) + self.arg_kinds.count(ARG_OPT)
+        self.type_args: list[TypeParam] | None = type_args
```

Comment:
This new field should be serialized.

Reply:
Not sure we actually need that here. Could be missing something but my understanding is that `type_args` is only used for the `SemanticAnalyzer` so there isn't a need to serialize it.

_If it were necessary, mypy would've probably crashed for me while testing the PR with Home Assistant._

Reply:
Yeah, this is just part of the syntax and doesn't get used beyond semantic analysis.